packages/amplify-util-uibuilder/src/__tests__/cloneComponentsFromEnv.test.ts

```diff
@@ -1,37 +1,37 @@
+jest.mock('aws-sdk', () => {
+  return {
+    AmplifyUIBuilder: jest.fn(() => {
+      return {
+        createComponent: () => ({
+          promise: () => true,
+        }),
+      };
+    }),
+  };
+});
+jest.mock('../commands/utils/extractArgs');
+import * as extractArgsDependency from '../commands/utils/extractArgs';
+jest.mock('../commands/utils/syncAmplifyUiBuilderComponents');
+import * as listUiBuilderComponentsDependency from '../commands/utils/syncAmplifyUiBuilderComponents';
+import { run } from '../commands/cloneComponentsFromEnv';
+
 describe('can clone components to new environment', () => {
   let context: any;
-  let aws: any;
-  let listUiBuilderComponents: any;
-  let extractArgs: any;
   beforeEach(() => {
     context = {
       amplify: {
         invokePluginMethod: () => true,
       },
     };
-    jest.mock('aws-sdk', () => {
-      return {
-        AmplifyUIBuilder: jest.fn(() => {
-          return {
-            createComponent: () => ({
-              promise: () => true,
-            }),
-          };
-        }),
-      };
-    });
-    jest.mock('../commands/utils/extractArgs');
-    jest.mock('../commands/utils/syncAmplifyUiBuilderComponents');
-    aws = require('aws-sdk');
-    extractArgs = require('../commands/utils/extractArgs').extractArgs;
-    listUiBuilderComponents = require('../commands/utils/syncAmplifyUiBuilderComponents').listUiBuilderComponents;
-    extractArgs.mockImplementation(() => ({
+    // @ts-ignore
+    extractArgsDependency.extractArgs = jest.fn().mockImplementation(() => ({
       sourceEnvName: 'sourceEnvName',
       newEnvName: 'newEnvName',
       appId: 'appId',
       environmentName: 'environmentName',
     }));
-    listUiBuilderComponents.mockImplementation((context: any, envName: any) => {
+    // @ts-ignore
+    listUiBuilderComponentsDependency.listUiBuilderComponents = jest.fn().mockImplementation((context: any, envName: any) => {
```

Comment:
not sure exactly which TS errors you were running into, but does adding this help?
```ts
const listUiBuilderComponentsDependency_mock = listUiBuilderComponentsDependency as jest.Mocked<typeof listUiBuilderComponentsDependency>;
```
Then instead of attaching the mocks to `listUiBuilderComponentsDependency`, attach them to `listUiBuilderComponentsDependency_mock` which _should_ be typed correctly. Here's one example: https://github.com/aws-amplify/amplify-cli/blob/e37718059cffdfecfe25a16ffdbf907cd2893949/packages/amplify-category-storage/src/__tests__/provider-utils/awscloudformation/cfn-template-utils.test.ts#L6-L9

Reply:
That doesn't work, I can type them as `any` to get rid of the ts-ignore annotations.